Feature/pop up window 8th Milestone

AmineDerbal opened the pull request:
In this milestone :

- removed the card projects from the HTML 👍 
- the card projects are created dynamically in the javascript when the page load 👍 
- the card project data are stored in an Object 👍 
- create the pop-up window 👍 
- add a close pop-up window for events in the pop-up close button and the shadow of the modal pop-up window 👍 
- in this Figma template there is only one image for the pop-up ℹ️ 
- I changed the technology list background colors in the popup-window for the contrast ℹ️

AmineDerbal committed:
add the card project dynamically from the javascript

AmineDerbal committed:
add popup style

AmineDerbal committed:
add popup

Yosefgeda committed:
PopUp window

AmineDerbal committed:
add close modal event

AmineDerbal committed:
add spacing in the functions

laiifuu commented on script.js:
```diff
@@ -2,15 +2,264 @@ const mobileMenu = document.querySelector('#mobile-menu-modal');
+const cardModel2Informations = [
+  'Profesional Art Printing Data',
+  "A daily selection of privately personalized reads; no accounts or sign-ups required. has been the industry's standard",
+  ['html', 'bootstrap', 'Ruby'],
+  'See Project',
+];
+const cardItem2Title = 'Data DashboardHealthcare';
+const cardItem3Title = 'Website Protfolio ';
+
+const projectsData = {
+  workModel1: {
+    title: cardModel1Informations[0],
+    description: cardModel1Informations[1],
+    techsList: cardModel1Informations[2],
+    project: cardModel1Informations[3],
+  },
+  workModel2: {
+    title: cardModel2Informations[0],
+    description: cardModel2Informations[1],
+    techsList: cardModel2Informations[2],
+    project: cardModel2Informations[3],
+  },
+  item2: {
+    title: cardItem2Title,
+  },
+  item3: {
+    title: cardItem3Title,
+  },
+  cardModel: {
+    classes: [
+      'work_model_1',
+      'work_model_2 item-1',
+      'work_model_2 item-2',
+      'work_model_2 item-3',
+      'work_model_2',
+      'work_model_2 item-2',
+      'work_model_2 item-3',
+    ],
+  },
+};
```
Your projects information should be organized in an array containing objects, one object for each project. It should look like something like this: 

```javascript
const projects = [{
    name: "Submarine game",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
}]
```
The array should hold a number of objects which is the same number of your projects. 
This will make your code cleaner.

laiifuu commented on script.js:
```diff
@@ -2,15 +2,264 @@ const mobileMenu = document.querySelector('#mobile-menu-modal');
+  modelProject.appendChild(addProjectButton(modelData));
+
+  return card;
+};
```
[OPTIONAL] I noticed for dynamically creating the project cards, you relied on the `document.createElement()` method, which is completely fine. You can consider using `Template literals` to create them instead, as it will make your code more legible, clearer and cleaner. You can refer to this [link](https://wesbos.com/template-strings-html) for more information about the topic.

laiifuu reviewed (changes requested):
# Status: Chanegs Required :recycle: 

## Hi team :wave: 
Good job so far with your first popup modals :clap: :clap: 
There are some issues that you still need to work on to go to the next project but you are almost there!


## Highlights
- [x] Nice use of Github Flow :heavy_check_mark: 
- [x] Descriptive pull request :heavy_check_mark: 

## Required Changes ♻️
- [ ] Nice job working on the popup window! However, as requested in the requirements, you should stick to the design as much as possible fot both the mobile and desktop views:
### Your design: 
#### Mobile
![Screen Shot 2023-01-31 at 20 53 10](https://user-images.githubusercontent.com/93520178/215867942-3a9ae831-b61f-4014-bd60-6c69a241a08a.png)
#### Desktop: 
[Screencast from 2023-01-31 08:21:10 PM.webm](https://user-images.githubusercontent.com/93520178/215865984-18ec3be8-d419-4388-bc39-a8da3f598c8c.webm)
### The template 
#### Mobile: 
![Screenshot from 2023-01-31 20-45-34](https://user-images.githubusercontent.com/93520178/215866399-57f79805-4747-43b0-8ab1-f1796016729d.png)
#### Desktop: 
![Screenshot from 2023-01-31 20-46-02](https://user-images.githubusercontent.com/93520178/215866486-8fe9227f-e382-4704-9484-8f118c13df65.png)

- [ ] When the popup appears, it is still possible to scroll down the page. This gives the user a bad experience when he's only trying to scroll down the popup itself. I suggest you fix this by adding a class with the `overflow: hidden;` declaration to the `body` whenever the the popup opens, and remove it when it closes. 

- [ ] In the desktop view, only the first two projects have the popup functionality, when all the projects should have it. I suggest adding the `See Project` button to the rest of the projects.
 
_Check the comments under the review :point_down:_

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and :computer: Happy coding :computer:

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

AmineDerbal committed:
fix popup windows design

AmineDerbal committed:
add overflow in popup

AmineDerbal committed:
display button project

AmineDerbal committed:
fix style errors

AmineDerbal commented:
- fixed the pop-up window design
- toggle overflow to hidden when the pop-up window is displayed
- display all button project
- for the optional I will implement them at a later date

AmineDerbal committed:
fix image dimension in pop-up window

AmineDerbal committed:
fix style errors

AmineDerbal committed:
fix image url

Shedrack-Sunday commented on script.js:
```diff
@@ -2,15 +2,295 @@ const mobileMenu = document.querySelector('#mobile-menu-modal');
 const closeModal = document.querySelector('#modal-close-icon img');
 const mobileMenuLinks = document.querySelectorAll('.mobile-menu-link');
+const cardWorks = document.getElementById('card_works');
+const cardModel1Informations = [
+  'Multi-Post Stories',
+  "A daily selection of privately personalized reads; no accounts or sign-ups required. This has been the industry's standard dummy text ever since the 1500s, when an unknown printer took a standard dummy text.",
+  ['css', 'html', 'bootstrap', 'Ruby'],
+  'See Project',
+];
+const cardModel2Informations = [
+  'Profesional Art Printing Data',
+  "A daily selection of privately personalized reads; no accounts or sign-ups required. has been the industry's standard",
+  ['html', 'bootstrap', 'Ruby'],
+  'See Project',
+];
+const cardItem2Title = 'Data DashboardHealthcare';
+const cardItem3Title = 'Website Protfolio ';
```
- [ ] Though these changes were suggested in the summary by the previous reviewer, I'm restating it here because you've not addressed them yet. 
- when the button to `see project` is clicked to pop up, there has to be a scroll by the user to see the other buttons which involved the live links and project source path. you'd agree with me that this will give a user a hard time navigating your site. as suggested by the previous reviewer, review your CSS code and avoid an overflow that can be set to `hidden`.

# YOUR MOBILE POP-UP VIEW
![image](https://user-images.githubusercontent.com/50990925/215983860-9e1f9a5e-73dd-44c3-9ec7-f68c5687e23d.png)

# FIGMA DESIGN FOR MOBILE POP-UP:
![image](https://user-images.githubusercontent.com/50990925/215983980-88548294-3d0a-4b05-88b6-5cf89fa8852d.png)


- [ ] For the desktop, there also seems to be an overlap between the source and buttons in relation to the image which the other reviewer also pointed out, and I strongly suggest that you fix it in line with the reasons I gave above for proper user navigation 👍 


# YOUR DESKTOP VIEW FOR POP-UP:
![image](https://user-images.githubusercontent.com/50990925/215984669-7e21ad70-69af-4944-9c2f-134138a26c3d.png)

# FIGMA DESKTOP POP-UP VIEW:
![image](https://user-images.githubusercontent.com/50990925/215984775-a5d28859-5316-4181-9cab-7d16e4d672ef.png)

AmineDerbal replied:
I've already fixed that before submitting

AmineDerbal committed:
fix techs list font size in mobile version

Shedrack-Sunday commented on script.js:
```diff
@@ -2,15 +2,295 @@ const mobileMenu = document.querySelector('#mobile-menu-modal');
+const cardItem2Title = 'Data DashboardHealthcare';
+const cardItem3Title = 'Website Protfolio ';
+
+const projectsData = {
+  workModel1: {
+    title: cardModel1Informations[0],
+    description: cardModel1Informations[1],
+    techsList: cardModel1Informations[2],
+    project: cardModel1Informations[3],
+  },
+  workModel2: {
+    title: cardModel2Informations[0],
+    description: cardModel2Informations[1],
+    techsList: cardModel2Informations[2],
+    project: cardModel2Informations[3],
```
- [ ] Also as pointed out by the previous reviewer, it seems your pop-up is only functional on two cards. this feature is meant to be on all cards, as you will get to add all your subsequent projects to this section. As seen in your project, it seems your previous project approval had an oversight on that feature. when a card is highlighted, it should provide the `See Project` feature on all cards. you did so in one card, and that's why you can't implement it. please add the same features for all :

# Your view with no button features for all cards:
![image](https://user-images.githubusercontent.com/50990925/215986147-2ee1afdc-d066-49f8-aa76-322954e1b2da.png)

# Figma design with sample of highlighted card that shows `see project` button:
![image](https://user-images.githubusercontent.com/50990925/215986294-46443a5b-ecc0-4b49-a5ad-3aeae21cf1a4.png)

AmineDerbal replied:
All the see projects are shown

Shedrack-Sunday commented on script.js:
```diff
@@ -2,15 +2,295 @@ const mobileMenu = document.querySelector('#mobile-menu-modal');
+    description: cardModel2Informations[1],
+    techsList: cardModel2Informations[2],
+    project: cardModel2Informations[3],
+  },
+  item2: {
+    title: cardItem2Title,
+  },
+  item3: {
+    title: cardItem3Title,
+  },
+  cardModel: {
+    classes: [
+      'work_model_1',
+      'work_model_2 item-1',
+      'work_model_2 item-2',
+      'work_model_2 item-3',
+      'work_model_2',
+      'work_model_2 item-2',
+      'work_model_2 item-3',
+    ],
```
- [ ] Lastly, I'll still refer to your previous reviewer and the project requirements. your card details should be organized in an array containing objects, one object for each project. then dynamically use that array to render your cards. 
here's a sample :
```
const projects = [{
    name: "Submarine game",
    description: "A Phaser3 platform game built as part of the Microverse JavaScript program.",
    image: "link_to_image.jpg",
    technologies: ["Phaser3", "JavaScript", "Jest"],
    link: "link_to_live_version.com",
}]
```

Shedrack-Sunday reviewed (changes requested):
Hi @AmineDerbal     👋 

# CHANGES REQUIRED ♻️

Great work on the project 👏
You've done well implementing the requirements,  However, there are a few you need to work on, to move to the next project. 💯 
![](https://user-images.githubusercontent.com/80612925/157397398-00904e26-60ec-46cd-9510-240661c250ef.gif)


Highlights
✅All linters are checked
✅ Descriptive commit messages 
  
Required Changes ♻️

- [ ] You did make efforts in the  Project and you really need a great job. However, there seems to be a little adjustments needed on both the desktop and  mobile view to make your whole code stand out an look good.

- [ ] the Reason for taking web accessibility seriously on both screen sizes is because it ensures that users with varying devices, including smaller screens such as smartphones, are able to access and use a website effectively. This includes individuals with disabilities who may use assistive technologies such as screen readers or magnifiers to access the web. By ensuring that websites are accessible on smaller viewports, developers can help make the web more inclusive for all users.  Kindly fix the margin on the right of the screen view. good job so far 👏 


_check the comment section for a detailed review_

Optional suggestions
Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application.


Cheers and Happy coding!👏👏👏
Feel free to leave any questions or comments in the PR thread if something is not 100% clear, and please remember to tag me in your question so I can receive the notification.


Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.

As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

AmineDerbal committed:
fix pop-up design

AmineDerbal committed:
fix objects in javascript

AmineDerbal commented:
https://www.loom.com/share/6d452fd1472c4059b424499770b18641

AmineDerbal commented:
- fixed pop-up window check video loom : https://loom.com/share/6d452fd1472c4059b424499770b18641
- fixed object array in Javascript
![mobile](https://user-images.githubusercontent.com/59117151/216013228-8d3f53f5-4b9b-4a91-bf43-7384f08076ed.png)

- add pointer to buttons in pop-up
- see live button will redirect to https://aminederbal.github.io/Portfolio/
- github button will redirect to https://github.com/AmineDerbal/Portfolio

AmineDerbal committed:
fix padding

AmineDerbal committed:
add pointer to pou-up button and redirection

osamaashraf6 reviewed (approved):
Hi @AmineDerbal , 
# STATUS: APPROVED! 🥇🟢🟢💯 
<img src="https://media3.giphy.com/media/xT0xezQGU5xCDJuCPe/giphy.gif"/>

Your project is complete!  There is nothing else to say other than... it's time to merge it : ship it: 
Congratulations! 🎉 

## Highlights💯🎈🪄✔️

- [x] Descriptive READme file in a professional way.✔️
- [x] Good usage in correcting GitHub flow.✔️
- [x] Nice design.✔️



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
